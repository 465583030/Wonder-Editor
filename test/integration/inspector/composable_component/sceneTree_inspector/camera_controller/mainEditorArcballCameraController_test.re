open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open MainEditorTransform;

let _ =
  describe("MainEditor ArcballCameraController", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test set currentSceneTreeNode to be camera", () => {
      beforeEach(() => {
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorSceneTool.setSceneFirstCameraToBeCurrentSceneTreeNode,
        );

        CurrentSelectSourceEditorService.setCurrentSelectSource(
          EditorType.SceneTree,
        )
        |> StateLogicService.getAndSetEditorState;
      });

      describe("test change arcballCameraController distance", () =>
        describe("test logic", () =>
          test("test change distance should set into engine", () => {
            MainEditorInspectorAddComponentTool.addArcballCameraControllerComponent();
            let currentGameObjectArcballCamera =
              GameObjectTool.getCurrentGameObjectArcballCamera();
            let value = 21.1;

            MainEditorArcballCameraControllerTool.changeDistanceAndBlur(
              ~cameraController=currentGameObjectArcballCamera,
              ~value,
              (),
            );

            ArcballCameraEngineService.unsafeGetArcballCameraControllerDistance(
              currentGameObjectArcballCamera,
            )
            |> StateLogicService.getEngineStateToGetData
            |. FloatService.truncateFloatValue(5)
            |> expect == value;
          })
        )
      );

      describe("test change arcballCameraController minDistance", () =>
        describe("test logic", () =>
          test("test change minDistance should set into engine", () => {
            MainEditorInspectorAddComponentTool.addArcballCameraControllerComponent();
            let currentGameObjectArcballCamera =
              GameObjectTool.getCurrentGameObjectArcballCamera();
            let value = 11.1;

            MainEditorArcballCameraControllerTool.changeMinDistanceAndBlur(
              ~cameraController=currentGameObjectArcballCamera,
              ~value,
              (),
            );

            ArcballCameraEngineService.unsafeGetArcballCameraControllerMinDistance(
              currentGameObjectArcballCamera,
            )
            |> StateLogicService.getEngineStateToGetData
            |. FloatService.truncateFloatValue(5)
            |> expect == value;
          })
        )
      );

      describe(
        "add shield dom for transformComponent if has arcballCameraController",
        () =>
        test("test snapshot for transform component", () => {
          MainEditorInspectorAddComponentTool.addArcballCameraControllerComponent();
          let currentGameObjectTransform =
            GameObjectTool.getCurrentSceneTreeNodeTransform();
          let component =
            BuildComponentTool.buildMainEditorTransformComponent(
              TestTool.buildEmptyAppState(),
              currentGameObjectTransform,
            );
          component |> ReactTestTool.createSnapshotAndMatch;
        })
      );
    });
  });