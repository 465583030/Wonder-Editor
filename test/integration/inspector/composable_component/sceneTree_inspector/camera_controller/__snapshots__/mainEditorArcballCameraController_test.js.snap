// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditor ArcballCameraController test set currentSceneTreeNode to be camera add shade dom for transformComponent if has arcballCameraController test snapshot for transform component 1`] = `
<article
  className="wonder-inspector-transform"
>
  <div
    className="transform-item"
  >
    <div
      className=""
    >
      position : 
    </div>
    <article
      className="wonder-transform-template"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="40"
        />
      </article>
    </article>
  </div>
  <div
    className="transform-item"
  >
    <div
      className=""
    >
      rotation : 
    </div>
    <article
      className="wonder-transform-template"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0"
        />
      </article>
    </article>
  </div>
  <div
    className="transform-item"
  >
    <div
      className=""
    >
      scale : 
    </div>
    <article
      className="wonder-transform-template"
    >
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          X : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Y : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
      <article
        className="wonder-float-input"
      >
        <span
          className="component-label"
        >
          Z : 
        </span>
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </article>
    </article>
  </div>
  <div
    className="transform-shade"
  />
</article>
`;
